app/(root)/(home)/page.tsx

```diff
@@ -1,25 +1,61 @@
 import { getUser } from "@/lib/actions/user.actions";
+import {
+  findPosts,
+  getAllPosts,
+  getFilteredPosts,
+} from "@/lib/actions/post.actions";
 
-import { User } from "@prisma/client";
-// import Posts from "../posts/page";
+import { CreateType, Post, User } from "@prisma/client";
+import Posts from "../../../components/post/Posts";
+import Pagination from "@/components/post/Pagination";
 
-export default async function Home() {
+export default async function Home({
+  searchParams,
+}: {
+  searchParams: { page: string; type: string };
+}) {
   const user = (await getUser()) as User;
 
+  const posts = await getAllPosts({
+    page: searchParams.page ?? "1",
+  });
```

Comment:
```suggestion
  const posts = await getAllPosts({
    page: searchParams.page ?? "1",
    searchTerm: searchParams.type as CreateType,
  });
```
Now that 'getAllPosts' handles both cases, we want to add the searchTerm here.

This works because it's optional in our code. If 'searchTerm' is undefined here, we just don't include it in our prisma query. That way we handle both cases at once :)